src/sentry/services/hybrid_cloud/organizationmember_mapping/__init__.py

```diff
@@ -0,0 +1,86 @@
+# Please do not use
+#     from __future__ import annotations
+# in modules such as this one where hybrid cloud service classes and data models are
+# defined, because we want to reflect on type annotations and avoid forward references.
+
+from abc import abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional, TypedDict
+
+from django.utils import timezone
+
+from sentry.models import OrganizationMember
+from sentry.services.hybrid_cloud import InterfaceWithLifecycle, silo_mode_delegation, stubbed
+from sentry.silo import SiloMode
+
+
+@dataclass(frozen=True, eq=True)
+class RpcOrganizationMemberMapping:
+    organization_id: int = -1
+    date_created: datetime = timezone.now()
+
+    role: str = ""
+    user_id: Optional[int] = None
+    email: Optional[str] = None
+    inviter_id: Optional[int] = None
+    invite_status: Optional[int] = None
+
+
+class RpcOrganizationMemberMappingUpdate(TypedDict):
```

Comment:
I believe you need to pass total=False to allow empty keys, see `class UserUpdateArgs(TypedDict, total=False):1`

Reply:
This may need test coverage?